src/client/src/Client.c

```diff
@@ -94,6 +94,64 @@ STATUS checkIntermittentProducerCallback(UINT32 timerId, UINT64 currentTime, UIN
     return retStatus;
 }
 
+STATUS configureClientWithRetryStrategy(PKinesisVideoClient pKinesisVideoClient) {
+    ENTERS();
+    PRetryStrategy pRetryStrategy = NULL;
+    STATUS retStatus = STATUS_SUCCESS;
+    KVS_RETRY_STRATEGY_TYPE defaultKvsRetryStrategyType = KVS_RETRY_STRATEGY_EXPONENTIAL_BACKOFF_WAIT;
+
+    CHK(pKinesisVideoClient != NULL, STATUS_NULL_ARG);
+
+    // If the callbacks for retry strategy are already set, then use that otherwise
+    // build the client with a default retry strategy.
+    if (pKinesisVideoClient->clientCallbacks.getRetryStrategyFn != NULL) {
+        CHK_STATUS(pKinesisVideoClient->clientCallbacks.getRetryStrategyFn(TO_CLIENT_HANDLE(pKinesisVideoClient)));
+    } else {
+        switch (defaultKvsRetryStrategyType) {

```

Comment:
This is an invariant and the static code analysis tools should have flagged this.